Backpack-SwiftUI/AppSearchModal/Classes/BPKAppSearchModal.swift

```diff
@@ -76,6 +75,19 @@ public struct BPKAppSearchModal: View {
         .background(.surfaceDefaultColor)
     }
     
+    private var searchInputSummary: some View {
+        BPKSearchInputSummary(
+            placeholder: inputHint,
+            clearAction: .init(accessibilityLabel: clearAction?.accessibilityLabel ?? "", action: {
+                inputText = ""
```

Comment:
I wonder if this whole action should be handled by the client, rather than from the component. As in, it should be the client the one that clears out this text field, not the BPK component.
Either this or just not expose a closure for an action, if its going to be the component that clears the text (I'd not recommend this).

Reply:
@frugoman thanks for your comment. In case of Date Selector we need to handle clear action on the client side, so we could perform some logic around date selections. However, we can transform it into "clearActionHandler", optional function, which would be called after clearing text out. And the clearing text would be left in the component implementation inside.
Otherwise, we need to keep it as it is.

Reply:
Right, I think then that this should be always in the client. even letting the client clear the text themselves. this releases this component of all logic around that and gives further flexibilty for the future (say for example, if we wanted to make the clear button be something else.)
another point for leaving this to the client is that now this component no longer has to 'guess' what to do with the inputState and can always be controlled by the client.

Reply:
That said, i'm not sure why you also removed the text field state. that mechanism is the one that controls the state of the field and restricted it to either be default or show the clear button.

Reply:
Basically, what happened is taking the clear action configuration out into the initialiser (it would be aligned with Android approach). And remove all unnecessary states (valid, error, disable). And we added a highlight state, so we could make borders highlighted even if the field is not in focus. This one is crucial for Date Selector latest design. Saying that, we can't have clear and highlight states at the same time.

Reply:
sorry, did you mean that we should not have highlight and clear at teh same time, or that we should and we cannot right now?

Reply:
btw, i think youre not setting the state/style of the search input field here, to handle the manual setting of the highlughted (focused) state

Reply:
We should and we cannot right now
In BPKAppSearchModal we don't need manual highlighting state